pkg/operator/certrotationcontroller/satokensigner_controller.go

```diff
@@ -274,6 +285,30 @@ func (c *SATokenSignerController) syncWorker() error {
 	return nil
 }
 
+// isValidKeyPair checks if a public/private keys match i.e if the tokens signed
+// with private key can be validated using public key. if "pubKeyData" or "privKeyData"
+// contains multiple keys, it tries to validate the first available keys in byte slices
+func isValidKeyPair(pubKeyData, privKeyData []byte) (bool, error) {
+	privKey, err := keyutil.ParsePrivateKeyPEM(privKeyData)
+	if err != nil {
+		return false, err
+	}
+	rsaPrivateKey, ok := privKey.(*rsa.PrivateKey)
```

Comment:
does it have to be RSA? do we enforce this elsewhere?

Reply:
afaik we only ever use RSA. It's baked in to how the tokens are generated.